File: src/main/java/com/server/ordering/repository/CouponRepository.java
package com.server.ordering.repository;

import com.server.ordering.domain.Coupon;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import javax.persistence.EntityManager;
import java.util.List;

@Repository
@RequiredArgsConstructor
public class CouponRepository {

    private final EntityManager em;

    public void save(Coupon coupon) {
        em.persist(coupon);
    }

    public Coupon find(String id) {
        return em.find(Coupon.class, id);
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/repository/CouponRepository.kt
package com.byeoru.ordering_server.repository

import com.byeoru.ordering_server.domain.Coupon
import org.springframework.stereotype.Repository
import javax.persistence.EntityManager
import javax.persistence.NoResultException

@Repository
class CouponRepository(private val em: EntityManager) {

    fun save(coupon: Coupon) = em.persist(coupon)

    @Throws(NoResultException::class)
    fun find(id: String): Coupon = em.find(Coupon::class.java, id)
}